### PR TITLE
feat(api-server): add list agent api

### PR DESCRIPTION
**Pull request by oojimmy**
# Describe your changes

1.  add list agent API expose to `/api/listAgent`
2. add cli agent list subcommand to get all agent info

## Issue
- [ ] If this PR fixes an issue, paste the link below

## Test:
- [x] What tests are done for this PR

### CODE REVIEW

File: src/api-server/http/api/api.go [hunk 1]
```diff
@@ -19,12 +19,14 @@ package api
 const (
 	ROOT            = "/api"
 	LIST_MODULE     = "/listModule"
+	LIST_AGENT      = "/listAgent"
```
Comment (nascentcore-eng) [inline src/api-server/http/api/api.go:25]: Should we use /api/agents path, and 'GET' method, which is more aligned with RESTFul convention?

File: src/api-server/http/module_manager.go [hunk 2]
```diff
@@ -152,6 +152,44 @@ func (mgr *ModuleManager) createModule(body CreateModuleReq) CreateModuleResp {
 	}}
 }
 
+// listAgentHttp godoc
+// @Summary      List all agent
+// @Description  List all agent
+// @Tags         agent
+// @Accept       json
+// @Produce      json
+// @Param			   fields	 query	string	false  "query field search like 'id,name,createTime'"
```
Comment (nascentcore-eng) [inline src/api-server/http/module_manager.go]: What's the meaning of each field here?

File: src/api-server/http/module_manager.go [hunk 3]
```diff
@@ -152,6 +152,44 @@ func (mgr *ModuleManager) createModule(body CreateModuleReq) CreateModuleResp {
...
+	fields, exists := c.GetQuery(fieldsKey)
+	if !exists {
+		log.Debugf("listModule request has no 'fields', use default fields: %s", defaultFields)
+		fields = defaultFields
+	}
+	result := mgr.listAgent(ListModuleReq{Fields: fields})
+	c.JSON(http.StatusOK, result)
+}
+
+func (mgr *ModuleManager) listAgent(req ListModuleReq) ListAgentResp {
```
Comment (nascentcore-eng) [inline src/api-server/http/module_manager.go]: Why use ListModuleReq as argument for listAgent?
You can use type alias to define a ListAgentReq instead.

File: src/api-server/http/module_manager_test.go [hunk 4]
```diff
@@ -292,3 +293,15 @@ func unDeployModule(t *testing.T, moduleID string, agentID string, r *gin.Engine
...
+	assert := assert.New(t)
+
+	r.GET("/api/listAgent", mgr.listAgentHttp)
+	req, _ := http.NewRequest("GET", "/api/listAgent", nil)
+	w := httptest.NewRecorder()
+	r.ServeHTTP(w, req)
+	resultStr := w.Body.String()
+	fmt.Printf("list agent: %s", resultStr)
+	assert.Contains(resultStr, agentID)
+}
```
Comment (nascentcore-eng) [inline src/api-server/http/module_manager_test.go]: Stop using t *testing.T in test helper, let this return error

**Review by nascentcore-eng — CHANGES_REQUESTED** [review]
There needs to be tests for ListAgent API

File: src/api-server/http/module_manager.go (hunk 2 above)
Comment (oojimmy) [reply src/api-server/http/module_manager.go]: this field contained database table column names, we can set these fields to determine which column will be selected and returned.

File: src/api-server/http/api/api.go (hunk 1 above)
Comment (oojimmy) [reply src/api-server/http/api/api.go:25]: add this todo comment

File: src/api-server/http/api/api.go [hunk 5]
```diff
@@ -19,14 +19,17 @@ import "strings"
 
 // API path components.
 const (
+	// todo: change to resful style
```
Comment (nascentcore-eng) [inline src/api-server/http/api/api.go:22]: TODO should have assignee:
And 
TODO(...):

**Review by nascentcore-eng — APPROVED** [review]

**Pull request merged**